src/OrchardCore.Modules/OrchardCore.ContentTypes/Services/StereotypeService.cs

```diff
@@ -1,27 +1,53 @@
+using System;
 using System.Collections.Generic;
+using System.Linq;
 using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using OrchardCore.ContentManagement;
+using OrchardCore.ContentManagement.Metadata.Models;
+using OrchardCore.Modules;
 
 namespace OrchardCore.ContentTypes.Services
 {
     public class StereotypeService : IStereotypeService
     {
         private readonly IEnumerable<IStereotypesProvider> _providers;
+        private readonly IContentDefinitionService _contentDefinitionService;
+        private readonly ILogger<StereotypeService> _logger;
 
-        public StereotypeService(IEnumerable<IStereotypesProvider> providers)
+        public StereotypeService(
+            IEnumerable<IStereotypesProvider> providers,
+            IContentDefinitionService contentDefinitionService,
+            ILogger<StereotypeService> logger)
         {
             _providers = providers;
+            _contentDefinitionService = contentDefinitionService;
+            _logger = logger;
         }
 
         public async Task<IEnumerable<StereotypeDescription>> GetStereotypesAsync()
         {
-            var descriptions = new List<StereotypeDescription>();
+            var providerStereotypes = (await _providers.InvokeAsync(provider => provider.GetStereotypesAsync(), _logger)).ToList();
 
-            foreach (var provider in _providers)
+            var stereotypes = providerStereotypes.Select(providerStereotype => providerStereotype.Stereotype)
+                .ToHashSet(StringComparer.OrdinalIgnoreCase);
+
+            foreach (var contentType in await _contentDefinitionService.GetTypesAsync())
             {
-                descriptions.AddRange(await provider.GetStereotypesAsync());
+                if (!contentType.TypeDefinition.TryGetStereotype(out var stereotype) ||
+                    stereotypes.Contains(stereotype))
+                {
+                    continue;
+                }
+
+                providerStereotypes.Add(new StereotypeDescription
+                {
+                    Stereotype = stereotype,
+                    DisplayName = stereotype
+                });
             }
 
-            return descriptions;
+            return providerStereotypes.OrderBy(x => x.DisplayName);
         }
```

Comment:
@MikeAlhayek   No problem. Ignore my previous comment, I had a problem with the original code, but after pulling your push the problem was fixed，thanks!